test/integration/app/wallets/invoice.spec.ts

```diff
@@ -20,45 +24,205 @@ import {
     const result = await walletInvoices.findByPaymentHash(getHash(request))
     if (result instanceof Error) throw result
-    const { walletId } = result
-    expect(walletId).toBe(walletId1)
+    const { walletId, currency } = result
+    expect(walletId).toBe(walletIdBtc)
+    expect(currency).toBe(WalletCurrency.Btc)
+
+    const decodedInvoice = decodeInvoice(request)
+    if (decodedInvoice instanceof Error) throw decodedInvoice
+
+    const { amount } = decodedInvoice
+    expect(amount).toBe(amountInput)
   })
 
   it("add a self generated invoice without amount", async () => {
     const lnInvoice = await Wallets.addInvoiceNoAmountByWalletId({
-      walletId: walletId1,
+      walletId: walletIdBtc,
+    })
+    if (lnInvoice instanceof Error) throw lnInvoice
+    const { paymentRequest: request } = lnInvoice
+
+    const result = await walletInvoices.findByPaymentHash(getHash(request))
+    if (result instanceof Error) throw result
+    const { walletId, currency } = result
+    expect(walletId).toBe(walletIdBtc)
+    expect(currency).toBe(WalletCurrency.Btc)
+  })
+
+  it("adds a public with amount invoice", async () => {
+    const amountInput = 10
+
+    const lnInvoice = await Wallets.addInvoiceForRecipient({
+      recipientWalletId: walletIdBtc,
+      amount: amountInput,
     })
     if (lnInvoice instanceof Error) throw lnInvoice
     const { paymentRequest: request } = lnInvoice
+    expect(request.startsWith("lnbcrt1")).toBeTruthy()
 
     const result = await walletInvoices.findByPaymentHash(getHash(request))
     if (result instanceof Error) throw result
-    const { walletId } = result
-    expect(walletId).toBe(walletId1)
+    const { walletId, selfGenerated } = result
+    expect(String(walletId)).toBe(String(walletIdBtc))
+    expect(selfGenerated).toBe(false)
+
+    const decodedInvoice = decodeInvoice(request)
+    if (decodedInvoice instanceof Error) throw decodedInvoice
+
+    const { amount } = decodedInvoice
+    expect(amount).toBe(amountInput)
   })
 
+  it("adds a public no amount invoice", async () => {
+    const lnInvoice = await Wallets.addInvoiceNoAmountForRecipient({
+      recipientWalletId: walletIdBtc,
+    })
+    if (lnInvoice instanceof Error) throw lnInvoice
+    const { paymentRequest: request } = lnInvoice
+    expect(request.startsWith("lnbcrt1")).toBeTruthy()
+
+    const result = await walletInvoices.findByPaymentHash(getHash(request))
+    if (result instanceof Error) throw result
+    const { walletId, selfGenerated } = result
+    expect(String(walletId)).toBe(String(walletIdBtc))
+    expect(selfGenerated).toBe(false)
+  })
+})
+
+describe("Wallet - addInvoice Fiat", () => {
+  it("add a self generated fiat invoice", async () => {
+    const price = await getCurrentPrice()
+    if (price instanceof Error) return price
+
+    const fiatInput = 100
+    const sats = fiatInput / price
+
+    const lnInvoice = await Wallets.addInvoiceByWalletId({
```

Comment:
I'm pretty sure this should be a different use case or? `addFiatInvoiceByWalletId`. Or is it the same mutation that creates both fiat and etc invoices?

Reply:
TDB. 

but that might be make sense to have 2 endpoints.

Reply:
actually not sure. the currency is embedded in the wallet, so there is no need to pass this independently. 

the client should call `addInvoice({walletId, amount})` and depending of the currency of the wallet, the amount would either be fiat or sats. 

this logic on knowing whether to know if the wallet should be a fiat wallet or btc wallet should be done on the client